toolchain/check/convert.cpp

```diff
@@ -1082,6 +1116,19 @@ static auto PerformBuiltinConversion(Context& context, SemIR::LocId loc_id,
              .type_inst_id = type_inst_id,
              .witnesses_block_id = lookup_result.inst_block_id()});
       }
+    } else {
+      // If impl lookup fails, don't keep looking for another way to convert.
+      // See https://github.com/carbon-language/carbon-lang/issues/5122.
+      // TODO: Move into `LookupImplWitness` so it can add notes explaining
+      // failure.
+      if (target.diagnose) {
+        DiagnoseConversionFailureToConstraintValue(context, loc_id, value_id,
+                                                   target.type_id);
+      }
+      // FIXME: If target.diagnose is false, perhaps we shouldn't return an
```

Comment:
Is this meant to be a TODO or is it meant to be addressed in this PR?

Reply:
I would like to resolve this FIXME in this PR.

Reply:
I have deleted the FIXME based on my better understanding from [our conversation on Discord](https://discord.com/channels/655572317891461132/655578254970716160/1356296819159728289).